### ISSUE TITLE
Lua error while hot-reloading spacebuild

### ISSUE THREAD

**Issue by dvdvideo1234** [spacebuild/spacebuild#158]
```
[ERROR] stack overflow
  1. istable - [C]:-1
   2. Merge - lua/includes/extensions/table.lua:84
    3. Merge - lua/includes/extensions/table.lua:87
     4. Merge - lua/includes/extensions/table.lua:87
      5. Merge - lua/includes/extensions/table.lua:87
       6. Merge - lua/includes/extensions/table.lua:87
        7. Merge - lua/includes/extensions/table.lua:87
         8. Merge - lua/includes/extensions/table.lua:87
          9. Merge - lua/includes/extensions/table.lua:87
           10. Merge - lua/includes/extensions/table.lua:87
            11. Merge - lua/includes/extensions/table.lua:87
             12. Merge - lua/includes/extensions/table.lua:87
              13. Merge - lua/includes/extensions/table.lua:87
               14. Merge - lua/includes/extensions/table.lua:87
                15. Merge - lua/includes/extensions/table.lua:87
                 16. Merge - lua/includes/extensions/table.lua:87

```
Consider the following source:
```lua
local a = {}
      a.b = a
local c = {}
      c.b = c
print("A-------------------")
table.Merge(c, a)
print("B-------------------")
```
This will print:
```
A-------------------
[ERROR] stack overflow
  1. istable - [C]:-1
   2. Merge - lua/includes/extensions/table.lua:84
    3. Merge - lua/includes/extensions/table.lua:87
......
```

**Comment by ApertureDevelopment** [comment]
Hello,

Where did you get that code snipet from? The code is trying to merge a table containing itself with another table containing itself, which causes a stack overflow when attempting to merge.

**Comment by dvdvideo1234** [comment]
I coded it myself. Somewhere in the SB pack there is an exact behavior like that. Yesterday I've codded [my Laser addon](https://github.com/dvdvideo1234/LaserSTool) and I needed to [hot reload the tool script](https://github.com/dvdvideo1234/LaserSTool/blob/main/lua/weapons/gmod_tool/stools/laseremitter.lua), which triggered the error. The last thing I've installed was SB and SBEP to do a [maglev with the ship parts](https://steamuserimages-a.akamaihd.net/ugc/1858306660067858577/089FDBE80FD50AE744A5947E0FD3896C87A78263/?imw=5000&imh=5000&ima=fit&impolicy=Letterbox&imcolor=%23000000&letterbox=false), so the most logical thing was moving it out of the addons folder and start the game again and it worked.

That's why I am pretty sure this behavior is triggered on hot-reload saving a tool script.

**Comment by ApertureDevelopment** [comment]
The thing is, Hot-Reloading in GMod only affects the file which has been saved. Which means that if you change a file, GMod only loads this particular file again, and not the entire server. So if you encounter this error when reloading your tool file, the error most likely comes from there

**Comment by dvdvideo1234** [comment]
This is incorrect. While I work on many of my addons I see logs and initializing activity form different addons that are not mine all the time. Good example for this is when I work on the track assembly I hot Ctrl+S on the tool file and reloads `CAP` and `NPC spawn platforms` being shown on the console. Those are GMAs, and it will be even harder to locate the code :) Which version of SB and SBEF can you recommend me to install to test this further. I cloned the repo. Maybe I've missed something... :+1:
